### DISCUSSION TITLE
CSV Import - Additional Attributes?

### DISCUSSION THREAD

**Discussion by billgraham5** [keepassxreboot/keepassxc#6312] [Q&A] [answered]
Does anyone know if there is a way to import a CSV file including all of the additional attributes for custom fields previously created in 1Password?

When I import a CSV, I can map up to 10 fields and the remaining are ignored.  

I am unable to import the opvault version of the database successfully because the custom field titles are not imported correctly.  I am looking for a work-around.

**Comment by droidmonkey** [comment, marked as answer]
There currently is no way to import additional attributes. You can, however, combine them in excel or similar product into the notes column.

**Reply by howdytom** [reply]
@droidmonkey Using the notes field as a workaround does turn out to be a solution for me. I would like migrate various data e.g. serial numbers, server IDs, warranty, into KeePassXC. Do you have any plans to support custom fields?

**Reply by droidmonkey** [reply]
Probably not for csv import. It is non scalable and challenges the import wizard (makes it very complex).

**Reply by howdytom** [reply]
@droidmonkey okay, thank you for your super-fast reply! 

The main question remains. How do you safe sensitive data without custom fields?

**Reply by droidmonkey** [reply]
I'm confused by your question. Within keepassxc when editing an entry, go to the advanced tab and you can add custom attributes at the top.

**Reply by howdytom** [reply]
@droidmonkey Sorry, you are right. 

Frankly, manually adding passwords data into custom fields is a pain. I do not want to copy + paste thousands of entries from notes into custom fields. This migration process lacks some convenience, though. I was hoping for some less time-consuming process. For a lot users this could be a big reason not switching to KeePassXC

**Reply by droidmonkey** [reply]
I am not sure which program you are coming from that drove you towards so many custom attributes. Is it from a spreadsheet that you now want to be in keepassxc? Certainly you can automate this using the CLI tools.

**Reply by howdytom** [reply]
Thanks for your support. I used to manage a lot of custom fields with a 32-bit password manager called Steel. Unfortunately it has been discontinued. Luckily I am able to export those fields as CSV file. I tried to simplify the exported spreadsheet but there are still some important custom attributes I have to keep and migrate.

**Reply by howdytom** [reply]
As a solution I have imported fields into Kdbx database using KeeWeb. Thanks for your support!
